### PR TITLE
SOF-6343: upgrade codemirror

### PR DESCRIPTION
**Pull request by unsigned6**

### CODE REVIEW

File: src/components/source_editor/CodeMirror.jsx [hunk 1]
```diff
@@ -34,7 +38,11 @@ class CodeMirror extends React.Component {
                 }}
                 onFocus={onFocus}
                 onBlur={onBlur}
-                options={options}
+                readOnly={readOnly}
+                lineNumbers={false}
+                theme={androidstudio}
```
Comment (azech-hqs) [inline src/components/source_editor/CodeMirror.jsx]: It seems the `darcula` theme which was used previously is also available via [www.npmjs.com/package/@uiw/codemirror-theme-darcula](https://www.npmjs.com/package/@uiw/codemirror-theme-darcula). Was there a specific reason for the change to `androidstudio`?

File: tests/cucumber/features/3d-editor/multiple-selection-cancel.feature
Comment (azech-hqs) [inline tests/cucumber/features/3d-editor/multiple-selection-cancel.feature:1]: The content here didn't really change except for a few spaces and newlines. Unless some changes are missing, could we use the unchanged file instead?

**Review by azech-hqs — COMMENTED** [review]
👍

File: src/components/source_editor/CodeMirror.jsx (hunk 1 above)
Comment (unsigned6) [reply src/components/source_editor/CodeMirror.jsx]: Yes I was looking at https://uiwjs.github.io/react-codemirror/#/theme/data/dracula and it was looking different, but now I see that there are `darcula` and `dracula`

File: tests/cucumber/features/3d-editor/multiple-selection-cancel.feature
Comment (unsigned6) [reply tests/cucumber/features/3d-editor/multiple-selection-cancel.feature:1]: These are redundant spaces. I think we could keep that reformating

**Review by timurbazhirov — APPROVED** [review]

**Pull request merged**